### PR TITLE
Raise process gas and add tests/

### PR DESCRIPTION
**Pull request by prestwich**
simplify the bridge test by moving useful functionality to the test bridge deploy

incidentally closees #561

### CODE REVIEW

File: solidity/optics-xapps/contracts/bridge/test/MockCore.sol [hunk 1]
```diff
@@ -1,30 +1,86 @@
...
+import {MerkleTreeManager} from "@celo-org/optics-sol/contracts/Merkle.sol";
+import {QueueManager} from "@celo-org/optics-sol/contracts/Queue.sol";
+
+import {Message} from "@celo-org/optics-sol/libs/Message.sol";
+import {MerkleLib} from "@celo-org/optics-sol/libs/Merkle.sol";
+import {QueueLib} from "@celo-org/optics-sol/libs/Queue.sol";
+
+// We reproduce a significant amount of logic from `Home` to ensure that
+// calling enqueue here is AT LEAST AS EXPENSIVE as calling it on home
+contract MockCore is MerkleTreeManager, QueueManager {
```
Comment (ErinHales) [inline solidity/optics-xapps/contracts/bridge/test/MockCore.sol:13]: We should just inherit from Home here and skip reproducing all this logic

File: typescript/optics-tests/test/bridge/bridge.test.ts [hunk 2]
```diff
@@ -22,56 +17,355 @@ const BRIDGE_MESSAGE_TYPES = {
 
 const typeToBytes = (type: number) => `0x0${type}`;
 
-describe('Bridge', async () => {
+describe.only('Bridge', async () => {
```
Comment (ErinHales) [inline typescript/optics-tests/test/bridge/bridge.test.ts]: remove .only

File: typescript/optics-tests/test/bridge/bridge.test.ts [hunk 3]
```diff
@@ -22,56 +17,355 @@ const BRIDGE_MESSAGE_TYPES = {
...
+        deploy.bridgeRouter!.send(
+          ethers.constants.AddressZero,
+          0,
+          12378,
+          `0x${'00'.repeat(32)}`,
+        ),
+      ).to.be.revertedWith('!remote');
+    });
+
+    it('remotely-originating asset roundtrip', async () => {
```
Comment (ErinHales) [inline typescript/optics-tests/test/bridge/bridge.test.ts:60]: Change to describe and break up test cases

File: typescript/optics-tests/test/bridge/bridge.test.ts [hunk 4]
```diff
@@ -22,56 +17,355 @@ const BRIDGE_MESSAGE_TYPES = {
...
+        TOKEN_VALUE,
+        deploy.remoteDomain,
+        deployerId,
+      );
+      await expect(badTx).to.be.revertedWith(
+        'ERC20: burn amount exceeds balance',
+      );
+    });
+
+    it('locally-originating asset roundtrip', async () => {
```
Comment (ErinHales) [inline typescript/optics-tests/test/bridge/bridge.test.ts:129]: Same, change to describe

File: typescript/optics-tests/test/bridge/bridge.test.ts (hunk 3 above)
Comment (prestwich) [reply typescript/optics-tests/test/bridge/bridge.test.ts:60]: will in next PR 👍

**Review by ErinHales — APPROVED** [review]

**Pull request merged**